### ISSUE TITLE
代码运行问题

### ISSUE THREAD

**Issue by xiaoding06** [tangjielong928/mqspn#2]
<img width="649" alt="87af6f2ad5704de78c7bad0783f1035" src="https://github.com/user-attachments/assets/63c47b7f-aa72-4391-bbe1-5468941073d9">
您好，我想问问您跑起来这个代码的时候有显示这个报错吗根据您的readme文件复制的那些数据，例如GMNER用到的数据使用的是它代码里面的Twitter_10000或者Twitter_10000_v2.0里面的xml数据作为images_annotation，然后Vinvl_detection_path路径里面使用的是您提供的下载里面是使用Vinvl识别的2015和2017文件，raw_images使用的是IJCAI2019_data里面的Twitter2015和Twitter2017所有的原始图片数据，然后按照requirements里面的环境配置，运行后报这个错误。希望您有看到的话可以帮忙看看是不是哪部分数据有问题需要去修改

**Comment by tangjielong928** [comment]
使用的是twitter-10000_v2.0的xml，麻烦提供完整的datasets下的截图，以及对应的config文件

**Comment by tangjielong928** [comment]
> <img alt="87af6f2ad5704de78c7bad0783f1035" width="649" src="https://private-user-images.githubusercontent.com/150208917/366501181-63c47b7f-aa72-4391-bbe1-5468941073d9.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjYwNzAzOTksIm5iZiI6MTcyNjA3MDA5OSwicGF0aCI6Ii8xNTAyMDg5MTcvMzY2NTAxMTgxLTYzYzQ3YjdmLWFhNzItNDM5MS1iYmUxLTU0Njg5NDEwNzNkOS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwOTExJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDkxMVQxNTU0NTlaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT0wYjYxZDQ5OGYyMmJhNWUwMTVmM2E0MzFiMjQ3NDJlNDZlNDY4MzU4MjVhZTMzMDY4MmE5YWE3YjEwNzBmYjM5JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.tW0is0pLoEtjnnRxfk0QuO3ax2QvyF4iXW7ZCXCUGDE"> 您好，我想问问您跑起来这个代码的时候有显示这个报错吗根据您的readme文件复制的那些数据，例如GMNER用到的数据使用的是它代码里面的Twitter_10000或者Twitter_10000_v2.0里面的xml数据作为images_annotation，然后Vinvl_detection_path路径里面使用的是您提供的下载里面是使用Vinvl识别的2015和2017文件，raw_images使用的是IJCAI2019_data里面的Twitter2015和Twitter2017所有的原始图片数据，然后按照requirements里面的环境配置，运行后报这个错误。希望您有看到的话可以帮忙看看是不是哪部分数据有问题需要去修改


使用的是twitter-10000_v2.0的xml，麻烦提供下完整的datasets下的截图，以及对应的config文件

**Comment by xiaoding06** [comment]
特别感谢您的回复，
<img width="211" alt="e131be44285f3c8df5c68f46a55ed33" src="https://github.com/user-attachments/assets/350a79a5-333b-471c-9809-9273cd75a3c4">
zhe'ge这个图片是我的datasets下面的截图，images_annotation是您说的twitter-10000-v2,0里面的xml文件，raw_images是IJCAI_2019_data里面的所有2015和2017的原始图片，Vinvl_detection_path里面的是2015和2017的vinvl_extract_36文件，其他的两个是您提供的已经有了的数据文件。下面是我的twitter_GMNER.conf文件：train_path = data/datasets/twitterGMNER/txt/samples.txt
valid_path = data/datasets/twitterGMNER/txt/samples.txt
types_path = data/datasets/twitterGMNER/twitter10k_types.json
# annotation path
xml_path = data/datasets/images_annotation
#detection model path
detection_path = data/datasets/Vinvl_detection_path
save_path = data/checkpoints/twitterGMNER
# raw image path
image_path = data/datasets/raw_images
# textual model
model_path = bert-base-cased
tokenizer_path = bert-base-cased
# vision model
vit_name = openai/clip-vit-base-patch32
train_batch_size = 8
epochs = 50
lr = 2e-05
lr_warmup = 0.05
weight_decay = 0.01
max_grad_norm = 1.0
# set preidction setting
match_solver = hungarian
# query Number
entity_queries_num = 60
type_loss = celoss
match_warmup_epoch = 0
nil_weight = -1.0
match_boundary_weight = 2.0
match_class_weight = 2.0
loss_boundary_weight = 2.0
loss_class_weight = 2.0
use_aux_loss = True
deeply_weight = same
repeat_gt_entities = 45
split_epoch = 5
local_rank = -1
world_size = -1
sampling_processes = 4
label = twitter10k_train
log_path = data/checkpoints/twitterGMNER
device_id = 2
model_type = MQSPN
cpu = False
eval_batch_size = 8
prop_drop = 0.5
freeze_transformer = False
no_overlapping = True
no_partial_overlapping = True
no_duplicate = True
cls_threshold = 0.0
boundary_threshold = 0.0
pos_size = 25
char_lstm_layers = 1
lstm_layers = 3
char_size = 50
char_lstm_drop = 0.2
use_glove = False
use_pos = False
use_char_lstm = False
# change to false if oom
use_lstm = False
pool_type = max
use_token_level_encoder = True
num_token_entity_encoderlayer = 5
candidate_regions_num = 15
mask_ent2tok = True
mask_tok2ent = False
mask_ent2ent = True
mask_entself = True
word_mask_ent2tok = True
word_mask_tok2ent = False
word_mask_ent2ent = True
word_mask_entself = True
use_entity_pos = True
last_layer_for_loss = 3
seed = 47  
非常感谢您的回复

**Comment by tangjielong928** [comment]
@xiaoding06  您的文件路径是正确的，尝试将config文件中的train_path = data/datasets/twitterGMNER/train.txtvalid_path = data/datasets/twitterGMNER/dev.txt

**Comment by tangjielong928** [comment]
@xiaoding06 您的数据文件路径是正确的，尝试将config文件中的路径修改为：train_path = data/datasets/twitterGMNER/train.txt
valid_path = data/datasets/twitterGMNER/dev.txt

**Comment by xiaoding06** [comment]
谢谢您，把config文件中的路径修改后能够正常运行起来了，后面如果用到eval.conf的话就把下面这个twitterGMNER_eval.conf：dataset_path = data/datasets/twitterGMNER/txt/samples.txt
config = configs/twitterGMNER_eval.conf
# checkpoint_path = data/checkpoints/twitter10k/twitter10k_train/best_0806/best_model/best_model.pth
checkpoint_path = [Your checkpoint_path]
types_path = data/datasets/twitterGMNER/twitter10k_types.json
# annotation path
xml_path = data/datasets/images_annotation
#detection model path
detection_path = data/datasets/Vinvl_detection_path
# raw image path
image_path = data/datasets/raw_images
# textual model
model_path = bert-base-cased
tokenizer_path = bert-base-cased
# vision model
vit_name = openai/clip-vit-base-patch32
local_rank = -1
world_size = -1
lowercase = False
sampling_processes = 4
label = twitter10k_eval
log_path = data/checkpoints/twitterGMNER
debug = False
device_id = 2
model_type = MQSPN
cpu = False
eval_batch_size = 8
prop_drop = 0.5
freeze_transformer = False
no_overlapping = True
no_partial_overlapping = True
no_duplicate = True
cls_threshold = 0.0
boundary_threshold = 0.0
pos_size = 25
char_lstm_layers = 1
lstm_layers = 3
char_size = 50
char_lstm_drop = 0.2
use_glove = False
use_pos = False
use_char_lstm = False
use_lstm = False
pool_type = max
use_token_level_encoder = True
num_token_entity_encoderlayer = 5
entity_queries_num = 60
candidate_regions_num = 15
mask_ent2tok = True
mask_tok2ent = False
mask_ent2ent = True
mask_entself = True
word_mask_ent2tok = True
word_mask_tok2ent = False
word_mask_ent2ent = True
word_mask_entself = True
use_entity_pos = True
last_layer_for_loss = 3
seed = 47
这里面的dataset_path = data/datasets/twitterGMNER/test.txt把里面的sample.txt修改为test.txt去评估，还有checkpoint_path就是训练生成的best_model路径就可以了。非常感谢您的回复哈

**Comment by tangjielong928** [comment]
> 谢谢您，把config文件中的路径修改后能够正常运行起来了，后面如果用到eval.conf的话就把下面这个twitterGMNER_eval.conf：dataset_path = data/datasets/twitterGMNER/txt/samples.txt config = configs/twitterGMNER_eval.conf
> 
> # checkpoint_path = data/checkpoints/twitter10k/twitter10k_train/best_0806/best_model/best_model.pth
> checkpoint_path = [Your checkpoint_path] types_path = data/datasets/twitterGMNER/twitter10k_types.json
> 
> # annotation path
> xml_path = data/datasets/images_annotation #detection model path detection_path = data/datasets/Vinvl_detection_path
> 
> # raw image path
> image_path = data/datasets/raw_images
> 
> # textual model
> model_path = bert-base-cased tokenizer_path = bert-base-cased
> 
> # vision model
> vit_name = openai/clip-vit-base-patch32 local_rank = -1 world_size = -1 lowercase = False sampling_processes = 4 label = twitter10k_eval log_path = data/checkpoints/twitterGMNER debug = False device_id = 2 model_type = MQSPN cpu = False eval_batch_size = 8 prop_drop = 0.5 freeze_transformer = False no_overlapping = True no_partial_overlapping = True no_duplicate = True cls_threshold = 0.0 boundary_threshold = 0.0 pos_size = 25 char_lstm_layers = 1 lstm_layers = 3 char_size = 50 char_lstm_drop = 0.2 use_glove = False use_pos = False use_char_lstm = False use_lstm = False pool_type = max use_token_level_encoder = True num_token_entity_encoderlayer = 5 entity_queries_num = 60 candidate_regions_num = 15 mask_ent2tok = True mask_tok2ent = False mask_ent2ent = True mask_entself = True word_mask_ent2tok = True word_mask_tok2ent = False word_mask_ent2ent = True word_mask_entself = True use_entity_pos = True last_layer_for_loss = 3 seed = 47 这里面的dataset_path = data/datasets/twitterGMNER/test.txt把里面的sample.txt修改为test.txt去评估，还有checkpoint_path就是训练生成的best_model路径就可以了。非常感谢您的回复哈

对的
